services/src/main/java/org/jboss/windup/web/services/service/AnalysisContextService.java

```diff
@@ -109,4 +116,69 @@ private void ensureSystemRulesPathsPresent(AnalysisContext analysisContext)
                             analysisContext.getRulesPaths().add(rulesPath);
                     });
     }
+
+    /**
+     * Returns the execution that was executed with given analysis context.
+     */
+    public WindupExecution getExecutionOfAnalysisContext(AnalysisContext analysisContext)
+    {
+        if (analysisContext == null)
+            throw new WindupException("AnalysisContext was null.");
+
+        List<WindupExecution> executions = this.entityManager.createQuery("SELECT exec FROM WindupExecution exec WHERE exec.analysisContext = :ctx ORDER BY exec.id DESC", WindupExecution.class).getResultList();
+        if (executions.size() > 1)
+        {
+            String execIds = executions.stream().map(exec -> "" + exec.getId()).collect(Collectors.joining(", "));
+            LOG.warning("Multiple executions have the same AnalysisContext #" + analysisContext.getId() + ": " + execIds);
+        }
+        return executions.isEmpty() ? null : executions.get(0);
+    }
+
+    /**
+     * Returns the default analysis context for the given project, that is, the context that has no execution assigned.
+     *
+     * @return If there are multiple, it returns the last one and deletes the rest (for robustness).
+     *         If there's none, returns null.
+     */
+    public AnalysisContext getDefaultProjectAnalysisContext(Long projectId)
+    {
+        return deleteDefaultContextOfProject(projectId, true);
```

Comment:
@jsight We discussed that. It's a cleanup task that deletes multiple default configs, should that happen, which is invalid, hence it can delete the older defaults. Why wouldn't this be triggered by GET? But I can put that anywhere else :)